Xamarin.PropertyEditing/ViewModels/PropertiesViewModel.cs

```diff
@@ -164,6 +204,38 @@ private PropertyViewModel GetViewModel (IPropertyInfo property)
 			return new StringPropertyViewModel (property, this.editors);
 		}
 
+		void FilterData ()
+		{
+			this.properties.Clear ();
+
+			UpdateProperties ();
+		}
+
+		List<PropertyViewModel> cachedProperties;
+		public List<PropertyViewModel> CachedProperties {
+			get {
+				if (cachedProperties == null) {
+					cachedProperties = Properties.ToList ();
+				}
+				return cachedProperties;
+			}
+		}
+
+		IEnumerable<IGrouping<string, PropertyViewModel>> cachedPropertiesGroupBy;
+		public IEnumerable<IGrouping<string, PropertyViewModel>> CachedPropertiesGroupBy {
+			get {
+				if (cachedPropertiesGroupBy == null) {
+					cachedPropertiesGroupBy = CachedProperties.GroupBy (arg => arg.Category);
+				}
+				return cachedPropertiesGroupBy;
+			}
+		}
+
+		void PropertiesViewModel_PropertyChanged (object sender, System.ComponentModel.PropertyChangedEventArgs e)
+		{
+			cachedProperties = null;
```

Comment:
We don't need to invalidate the cache any time _any_ property changes. What we want is the original property list built by `UpdateProperties` stored safely, and then a second cached list that is a filtered view of the original list. `Properties` should expose The filtered list (and grouped list) are what we want to cache and invalidate whenever the original list changes, or the filter changes.